mlflow/server/js/src/experiment-tracking/pages/prompts/components/ModelConfigForm.tsx

```diff
@@ -0,0 +1,216 @@
+import {
+  FormUI,
+  InfoSmallIcon,
+  Popover,
+  RHFControlledComponents,
+  Typography,
+  useDesignSystemTheme,
+} from '@databricks/design-system';
+import { useFormContext } from 'react-hook-form';
+import { FormattedMessage, useIntl } from 'react-intl';
+
+export const ModelConfigForm = () => {
+  const { theme } = useDesignSystemTheme();
+  const intl = useIntl();
+  const {
+    control,
+    formState: { errors },
+  } = useFormContext();
+
+  const getFieldName = (name: string) => `modelConfig.${name}`;
+
+  /**
+   * Gets validation error for a model config field.
+   * Errors are stored as errors.modelConfig.fieldName by the parent form validation.
+   */
+  const getError = (name: string) => {
+    return (errors?.['modelConfig'] as any)?.[name];
+  };
+
+  return (
+    <div
+      css={{
+        border: `1px solid ${theme.colors.border}`,
+        borderRadius: theme.general.borderRadiusBase,
+        padding: theme.spacing.md,
+      }}
+    >
+      <div css={{ display: 'flex', alignItems: 'center', gap: theme.spacing.xs, marginBottom: theme.spacing.sm }}>
+        <Typography.Title level={4} css={{ marginBottom: 0 }}>
+          <FormattedMessage
+            defaultMessage="Model Configuration"
+            description="Section header for model configuration in prompt creation"
+          />
+        </Typography.Title>
+        <Popover.Root componentId="mlflow.prompts.model_config.help">
+          <Popover.Trigger aria-label="Model configuration help" css={{ border: 0, background: 'none', padding: 0 }}>
+            <InfoSmallIcon />
+          </Popover.Trigger>
+          <Popover.Content align="start">
+            <FormattedMessage
+              defaultMessage="Model configuration stores the LLM settings associated with this prompt."
+              description="Help text explaining model configuration purpose"
+            />
+            <Popover.Arrow />
+          </Popover.Content>
+        </Popover.Root>
+      </div>
+
+      <div css={{ display: 'flex', flexDirection: 'column', gap: theme.spacing.md }}>
+        <div>
+          <FormUI.Label htmlFor="mlflow.prompts.model_config.provider">
+            <FormattedMessage defaultMessage="Provider" description="Label for model provider input" />
+          </FormUI.Label>
+          <RHFControlledComponents.Input
```

Comment:
Can we add a TODO comment that we'll change this into a select box? Or maybe we can do that in this PR since https://github.com/mlflow/mlflow/pull/19009 has been merged.